consensus/hotstuff/timeoutaggregator/timeout_collectors.go

```diff
@@ -70,8 +74,13 @@ func (t *TimeoutCollectors) GetOrCreateCollector(view uint64) (hotstuff.TimeoutC
 		return clr, false, nil
 	}
 	t.collectors[view] = collector
+	if t.newestViewCachedCollector < view {
+		t.newestViewCachedCollector = view
+	}
 	t.lock.Unlock()
 
+	// report metrics outside lock and accept the fact that we might report not the same value observed in critical section.
```

Comment:
`t.lowestRetainedView` and `t.newestViewCachedCollector` are written and accessed in a concurrent manner and protected by `t.lock`
However, here, we are accessing the values outside of a lock. While it is for metrics purposes only, it still would be great to capture the atomic state at before we unlock in [line 80](https://github.com/onflow/flow-go/blob/66e1ffaa284b7ab3341539a8ccbb6cafc7866395/consensus/hotstuff/timeoutaggregator/timeout_collectors.go#L80).


_Suggestion_:
* While still within the lock, I would store all 3 values that we feed into the metrics here. 
* then we unlock and call into the `module.HotstuffMetrics` outside the lock for performance reasons.
 
I think this would be a good compromise, because then we capture the state at the end of the atomic operation. 

Specifically, I would suggest to replace [lines 80-83](https://github.com/onflow/flow-go/blob/66e1ffaa284b7ab3341539a8ccbb6cafc7866395/consensus/hotstuff/timeoutaggregator/timeout_collectors.go#L80-L83) by 
```golang
// Capture metrics values at the end of the atomic update. For performance reasons,
// The actual call to the `metrics` collector happens outside the lock (below)
lowestRetainedView := t.lowestRetainedView
newestViewCachedCollector := t.newestViewCachedCollector
activeCollectors := len(t.collectors)
t.lock.Unlock()

// report metrics outside lock and accept the fact that we might report not the same value observed in critical section.
t.metrics.TimeoutCollectorsRange(lowestRetainedView, newestViewCachedCollector, activeCollectors)
```

Reply:
Honestly,  I don't remember why I did it this way, maybe just wanted to avoid introducing new variables for such case. Probably my reasoning was that even if we don't report latest value it's not a problem since it will be reported by other goroutine.
I will update it so we report always newest value.